### ISSUE TITLE
Switch+scheduler problems

### ISSUE THREAD

**Issue by Arceliar** [yggdrasil-network/yggdrasil-go#555]
tl;dr we need to do something other than https://github.com/yggdrasil-network/yggdrasil-go/pull/542/commits/80ba24d51255c3751e2b25aceee52b20d59ff746 to fix the switch. That commit makes the network completely unusable if the shortest path (in hop count) is bad (e.g. anyone with a Tor peer or nodes on the edge of a wireless range).

What tends to happen is that multiple packets are queued up for the switch to process before the scheduler decides to run the switch. The switch then processes them all in sequence before anything else in the program has a chance to react. So the switch will send the first packet to the best link, and then all subsequent packets either get sent to a worse link or are dropped into a buffer. We want that to happen *if* the best link is blocked on a syscall, but what actually happens is that the switch processes all of these messages before the scheduler gets around to having the link attempt to send the first packet.

To fix this, we'll need an async algorithm to detect when a link is *really* blocked, communicate that information to the switch, and *not* make the switch dump all traffic into a bad link or overflow a buffer the moment it receives that block notification (because the link may have unblocked before the switch processes the blocked message -- in fact, I think that would *almost always* happen, given how the scheduler seems to run things).

**Comment by Arceliar** [comment]
Partly mitigated in later v0.3 releases, a better workaround will be done for v0.4 most likely.
